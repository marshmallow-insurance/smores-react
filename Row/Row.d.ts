import { FC } from 'react';
import { Color } from '../theme';
import { MarginProps } from '../utils/space';
export type RowProps = {
    iconLeft?: string;
    iconLeftColor?: Color;
    iconRight?: string;
    iconRightColor?: Color;
    handleClick?: () => void;
    heading: string;
    subHeading?: string;
    type?: 'first' | 'last' | 'curved';
    borderTop?: boolean;
    borderBottom?: boolean;
    boldHeading?: boolean;
} & MarginProps;
export declare const Row: FC<RowProps>;
