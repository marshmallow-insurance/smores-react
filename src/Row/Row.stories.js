import React from 'react'
import { Row } from '../Row'

export default {
  title: 'Row',
  conmponent: Row,
}

const Template = (args) => <Row {...args} />

export const Default = Template.bind({})

Default.args = {
  icon: 'info',
  heading: 'A lovely heading',
  subHeading: 'The quick brown fox jumps over the lazy dog',
  caret: true,
  type: 'curved',
  borderTop: true,
  borderBottom: true,
}

export const NoIcon = Template.bind({})

NoIcon.args = {
  heading: 'A lovely heading',
  subHeading: 'A row without an Icon',
}

export const NoSubheading = Template.bind({})

NoSubheading.args = {
  heading: 'A lovely heading without a subheading',
  icon: 'info',
}

export const NoCaret = Template.bind({})

NoCaret.args = {
  heading: 'A lovely heading',
  subHeading: 'A row without a caret',
  icon: 'info',
  caret: false,
}

export const First = Template.bind({})

First.args = {
  heading: 'The first row',
  subHeading: 'A first row with curved corners at the top and no bottom border',
  type: 'first',
  borderBottom: false,
}

export const Middle = Template.bind({})

Middle.args = {
  heading: 'A middle row',
  subHeading: `A row with angular corners to go between 'first' and 'last' rows`,
}

export const Last = Template.bind({})

Last.args = {
  heading: 'The last row',
  subHeading: 'A last row with curved corners at the bottom and no top border',
  type: 'last',
  borderTop: false,
}

export const Curved = Template.bind({})

Curved.args = {
  heading: 'A curved row',
  type: 'curved',
  subHeading: 'A curved row with a border radius of 8px on all corners',
}
