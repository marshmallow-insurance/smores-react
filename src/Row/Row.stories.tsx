import React from 'react'
import { Row, RowProps } from './Row'

export default {
  title: 'Row',
  component: Row,
  argTypes: { handleClick: { action: 'clicked' } },
}

const Template = (props: RowProps) => <Row {...props} />

export const Default = Template.bind({})

Default.args = {
  iconLeft: 'info',
  iconLeftColor: 'liquorice',
  iconRight: 'caret',
  iconRightColor: 'subtext',
  heading: 'A lovely heading',
  subHeading: 'The quick brown fox jumps over the lazy dog',
  type: 'curved',
  borderTop: true,
  borderBottom: true,
}

export const NoIcons = Template.bind({})

NoIcons.args = {
  heading: 'A lovely heading',
  subHeading: 'A row without an IconLeft or IconRight',
}

export const NoSubheading = Template.bind({})

NoSubheading.args = {
  heading: 'A lovely heading without a subheading',
  iconLeft: 'info',
  iconRight: 'download',
}

export const BoldHeading = Template.bind({})

BoldHeading.args = {
  heading: 'A row with a bold heading',
  subHeading: `A row with a heading that has a font weight of 500`,
  boldHeading: true,
}

export const First = Template.bind({})

First.args = {
  heading: 'The first row',
  subHeading: 'A first row with curved corners at the top and no bottom border',
  type: 'first',
  borderBottom: false,
}

export const Middle = Template.bind({})

Middle.args = {
  heading: 'A middle row',
  subHeading: `A row with angular corners to go between 'first' and 'last' rows`,
}

export const Last = Template.bind({})

Last.args = {
  heading: 'The last row',
  subHeading: 'A last row with curved corners at the bottom and no top border',
  type: 'last',
  borderTop: false,
}

export const Curved = Template.bind({})

Curved.args = {
  heading: 'A curved row',
  type: 'curved',
  subHeading: 'A curved row with a border radius of 8px on all corners',
}
