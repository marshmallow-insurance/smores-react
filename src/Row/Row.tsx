import React, { FC } from 'react'
import styled, { css } from 'styled-components'

import { Box } from '../Box'
import { Icon } from '../Icon'
import { Icons } from '../Icon/iconsList'
import { Text } from '../Text'
import { Color, theme } from '../theme'
import { MarginProps } from '../utils/space'

export type RowProps = {
  iconLeft?: Icons
  iconLeftColor?: Color
  iconRight?: Icons
  iconRightColor?: Color
  handleClick?: () => void
  heading: string
  subHeading?: string
  type?: 'first' | 'last' | 'curved'
  borderTop?: boolean
  borderBottom?: boolean
  boldHeading?: boolean
} & MarginProps

export const Row: FC<RowProps> = ({
  iconLeft,
  iconLeftColor = 'liquorice',
  iconRight,
  iconRightColor = 'marzipan',
  handleClick,
  heading,
  subHeading,
  type,
  borderTop = true,
  borderBottom = true,
  boldHeading,
  ...marginProps
}) => {
  const windowWidth = screen.width

  return (
    <Container
      type={type}
      iconLeft={iconLeft}
      borderTop={borderTop}
      borderBottom={borderBottom}
      iconRight={iconRight}
      onClick={handleClick}
      boldHeading={boldHeading}
      {...marginProps}
    >
      {iconLeft && (
        <Icon
          render={iconLeft}
          size={windowWidth > 768 ? 24 : 18}
          color={iconLeftColor}
        />
      )}
      <Box>
        <Text tag="h1" typo="base">
          {heading}
        </Text>
        <Text tag="p" typo="desc-small" color="sesame">
          {subHeading}
        </Text>
      </Box>
      {iconRight && (
        <Icon
          className="iconRight"
          render={iconRight}
          size={24}
          rotate={iconRight === 'caret' ? -90 : 0}
          color={iconRightColor}
        />
      )}
    </Container>
  )
}

interface IContainer {
  type?: 'first' | 'last' | 'curved'
  iconLeft?: string
  iconRight?: string
  borderTop: boolean
  borderBottom: boolean
  boldHeading?: boolean
}

const Container = styled(Box)<IContainer>(
  ({ type, iconLeft, borderTop, borderBottom, boldHeading }) => css`
    border-radius: ${(type === 'first' && `16px 16px 0 0`) ||
    (type === 'curved' && `16px`) ||
    (type === 'last' && `0 0 16px 16px`) ||
    0};

    border-top: ${borderTop && type !== 'curved' && type !== 'first'
      ? `1px solid ${theme.colors.oatmeal}`
      : 'none'};
    border-bottom: ${borderBottom && type !== 'curved' && type !== 'last'
      ? `1px solid ${theme.colors.oatmeal}`
      : 'none'};

    background-color: ${theme.colors.custard};
    display: grid;
    grid-template-columns: ${iconLeft ? `24px auto 24px` : `auto 24px`};
    padding: 16px;
    grid-gap: 16px;
    align-items: center;

    cursor: pointer;

    h1 {
      font-weight: ${boldHeading && 500};
    }

    .iconRight {
      justify-self: end;
    }

    @media (min-width: 768px) {
      grid-gap: 24px;
    }
  `,
)
