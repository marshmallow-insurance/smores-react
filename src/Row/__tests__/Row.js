import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'

import { Row } from '../Row'

it('default renders correctly', () => {
  const { container } = render(<Row heading="This is a Row with a heading" />)

  expect(container).toMatchSnapshot()
})
