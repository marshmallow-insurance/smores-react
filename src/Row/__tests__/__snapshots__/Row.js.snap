// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default renders correctly 1`] = `
.c1 {
  font-size: 14px;
  line-height: 23px;
  font-weight: 400;
  text-align: left;
  cursor: inherit;
  color: #292924;
}

.c2 {
  font-size: 12px;
  line-height: 16px;
  font-weight: 400;
  text-align: left;
  cursor: inherit;
  color: #636768;
}

.c0 {
  border-radius: 0;
  border-top: 1px solid #DAD2C4;
  border-bottom: 1px solid #DAD2C4;
  background-color: #F1E9DC;
  display: grid;
  grid-template-columns: auto 24px;
  padding: 16px;
  grid-gap: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c0 .iconRight {
  justify-self: end;
}

@media (min-width:768px) {
  .c1 {
    font-size: 16px;
    line-height: 26px;
  }
}

@media (min-width:768px) {
  .c2 {
    font-size: 14px;
    line-height: 18px;
  }
}

@media (min-width:768px) {
  .c0 {
    grid-gap: 24px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class=""
    >
      <h1
        class="c1"
        color="liquorice"
        cursor="inherit"
        title=""
      >
        This is a Row with a heading
      </h1>
      <p
        class="c2"
        color="sesame"
        cursor="inherit"
        title=""
      />
    </div>
  </div>
</div>
`;
